Add native SparseH support for adjoint (#72)
PennyLaneAI/pennylane-lightning-gpu: .github/CHANGELOG.md

### New features since last release

* Explicit support for `qml.SparseHamiltonian` using the adjoint gradient method. 
  [(#72)](https://github.com/PennyLaneAI/pennylane-lightning-gpu/pull/72)

  This support allows users to explicitly make use of `qml.SparseHamiltonian` in expectation value calculations, and ensures the gradients can be taken efficiently. 

>   A user can now explicitly decide whether to decompose the Hamiltonian into separate Pauli-words, with evaluations happening over multiple GPUs, or convert the Hamiltonian directly to a sparse representation for evaluation on a single GPU. Depending on the Hamiltionian structure, a user may benefit from one method or the other.

Reviewer: ```suggestion
  A user can now explicitly decide whether to decompose the Hamiltonian into separate Pauli-words, with evaluations happening over multiple GPUs, or convert the Hamiltonian directly to a sparse representation for evaluation on a single GPU. Depending on the Hamiltonian structure, a user may benefit from one method or the other.
```

>   The work-flow for decomposing a Hamiltonian is as:

Reviewer: ```suggestion
  The workflow for decomposing a Hamiltonian is as:
```

  ```python
  obs_per_gpu = 1
  dev = qml.device("lightning.gpu", wires=num_wires, batch_obs=obs_per_gpu)

  H = sum([0.5*(i+1)*(qml.PauliZ(i)@qml.PauliZ(i+1)) for i in range(0, num_wires-1, 2)])

  @qml.qnode(dev, diff_method="adjoint")
  def circuit(params):
      for i in range(num_wires):
          qml.RX(params[i], i)
      return qml.expval(H)
  ```

>   For the new `qml.SparseHamiltonian` the above script becomes:

Author: ```suggestion
  For the new `qml.SparseHamiltonian` support, the above script becomes:
```
